sql/core/src/test/scala/org/apache/spark/sql/ResolveDefaultColumnsSuite.scala

```diff
@@ -256,4 +256,11 @@ class ResolveDefaultColumnsSuite extends QueryTest with SharedSparkSession {
       checkAnswer(sql("select * from t"), Row(1, "apache", "spark "))
     }
   }
+
+  test("SPARK-46958: Fix bug when canUpcast and result non-foldable expression") {
+    sql(s"CREATE TABLE t(key int, c STRING DEFAULT date '2018-11-17') " +
```

Comment:
does it work if the default expression is `CAST(date '2018-11-17' AS STRING)`? I think we should find the root cause and understand why the casted expression was considered as non-foldable.

Reply:
The case for the date type seems to be restored on the master branch, but ts values still have an issue caused by missing timeZone, thus, unresolved.